diktat-maven-plugin/src/main/kotlin/org/cqfn/diktat/plugin/maven/DiktatBaseMojo.kt

```diff
@@ -233,26 +228,23 @@ abstract class DiktatBaseMojo : AbstractMojo() {
             }
     }
 
-    private fun checkFile(file: File,
-                          lintErrors: MutableList<LintError>,
-                          baselineErrors: List<LintError>,
-                          ruleSets: Iterable<RuleSet>
+    private fun checkFile(
+        file: File,
+        lintErrors: MutableList<LintError>,
+        baselineErrors: List<LintError>
     ) {
-        val text = file.readText()
-        val params =
-            KtLint.ExperimentalParams(
-                fileName = file.absolutePath,
-                text = text,
-                ruleSets = ruleSets,
-                script = file.extension.equals("kts", ignoreCase = true),
-                cb = { lintError, isCorrected ->
-                    if (!baselineErrors.containsLintError(lintError)) {
-                        reporterImpl.onLintError(file.absolutePath, lintError, isCorrected)
-                        lintErrors.add(lintError)
-                    }
-                },
-                debug = debug
-            )
-        runAction(params)
+        val command = DiktatProcessCommand.Builder()
+            .file(file)
+            .fileContent(file.readText(Charsets.UTF_8))
+            .isScript(file.extension.equals("kts", ignoreCase = true))
+            .callback { error: DiktatError, isCorrected: Boolean ->
+                val ktlintError = error.unwrap()
+                if (!baselineErrors.containsLintError(ktlintError)) {
+                    reporterImpl.onLintError(file.absolutePath, ktlintError, isCorrected)
+                    lintErrors.add(ktlintError)
+                }
+            }
+            .build()
+        runAction(command)
```

Comment:
Where has the `debug` flag gone?

Reply:
introduced DiktatLogLevel with values DEBUG and INFO for now.
added as enum to support different log level